src/pfe/portal/routes/projectTypes.route.js

```diff
@@ -15,15 +15,97 @@ const Logger = require('../modules/utils/Logger');
 const router = express.Router();
 const log = new Logger(__filename);
 
+function sanitizeProjectType(array, type) {
+
+  // doesn't even have the expected fields, no-op return
+  if (!type.projectType || !type.projectSubtypes || !Array.isArray(type.projectSubtypes.items))
+    return array;
+
+  const sanitized = {
+    projectType: String(type.projectType),
+    projectSubtypes: {
+      prompt: String(type.projectSubtypes.prompt),
+      items: []
+    }
+  };
+
+  for (const item of type.projectSubtypes.items) {
+    if (!item.id)
+      continue;
+    sanitized.projectSubtypes.items.push({
+      id: String(item.id),
+      version: String(item.version),
+      label: String(item.label || item.id),
+      description: String(item.description)
+    });
+  }
+
+  if (sanitized.projectSubtypes.items.length > 0)
+    array.push(sanitized);
+
+  return array
+}
+
+async function getProjectTypes(provider) {
+  
+  const projectTypes = [];
+
+  // get projectTypes from extension provider
+  if (provider && typeof provider.getProjectTypes == 'function') {
+    const types = await provider.getProjectTypes();
+    if (Array.isArray(types))
+      types.reduce(sanitizeProjectType, projectTypes);
+  }
+
+  return projectTypes;
+}
+
+function addLanguage(projectType, language) {
+  if (!projectType.projectSubtypes.items.find(item => item.id == language))
+    projectType.projectSubtypes.items.push({ id: language, label: language });
```

Comment:
I printed out the output of template.label and template.language before this method call and saw output like:
language = java, label = Lagom Java. Should the addLanguage method also have the label passed into it as well? If the label is suppose to be the language, maybe add a comment explaining why, since it is a bit confusing.

Reply:
I will add a comment.  The label (`Lagom`) is on the parent level project type, and the properties for the subtypes here is the same on purpose

Reply:
Actually I misspoke.  The template's label is actually not used here in this API, it is used when we present the list of templates to user for project creation workflow.  In any case I'll add a comment